setup/classes/class.ilCtrlStructureReader.php

```diff
@@ -180,12 +175,17 @@ function read($a_cdir)
 											$ilDB->manipulate("DELETE FROM ctrl_calls WHERE comp_prefix IS NULL");
 										}
 										
-										$this->err_object->raiseError(
-											sprintf($lng->txt("duplicate_ctrl"),
+										require_once("./Services/Utilities/classes/class.ilUtil.php");
+										\ilUtil::sendFailure(
+											sprintf(
+												$lng->txt("duplicate_ctrl"),
 												$parent,
 												$this->class_script[$parent],
-												$a_cdir."/".$file)
-											, $this->err_object->MESSAGE);
+												$a_cdir."/".$file
+											),
+											true
+										);
+										\ilUtil::redirect("setup.php?cmd=clientlist");
```

Comment:
I'm not sure that this is correct. This might as well be called from somewhere else, like the DB-Update page. Please look for usage sides of `ilCtrlStructureReader`.